app/models/vmdb_database/seeding.rb

```diff
@@ -42,11 +41,16 @@ def db_server_ipaddress
       host
     end
 
-    def db_disk_size(disk)
-      MiqSystem.disk_usage(disk).first[:filesystem]
-    rescue RuntimeError => err
-      return nil if err.message.include?("does not exist")
-      raise
+    def data_disk_name(disk)
+      if disk && EvmDatabase.local?
+        require 'sys-filesystem'
+        begin
+          mount_point = Sys::Filesystem.mount_point(disk)
```

Comment:
This is failing for some people in development...

```
Errno::EACCES: Permission denied @ rb_file_s_stat - /var/lib/pgsql/data
```

Reply:
For me, it's working:

```
$ ls /var/lib/postgresql/11/main
ls: cannot open directory '/var/lib/postgresql/11/main': Permission denied
$ ls /var/lib/postgresql/11/
main

> Sys::Filesystem.mount_point('/var/lib/postgresql/11/main')
=> "/"
```

**But**, if I `chmod 700 /var/lib/postgresql/11`:

```
$ ls /var/lib/postgresql/11/main
ls: cannot open directory '/var/lib/postgresql/11/main': Permission denied
$ ls /var/lib/postgresql/11/
ls: cannot open directory '/var/lib/postgresql/11/': Permission denied

> Sys::Filesystem.mount_point('/var/lib/postgresql/11/main')
Traceback (most recent call last):
       16: from /home/himdel/.rbenv/versions/2.5.3/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
       15: from /home/himdel/.rbenv/versions/2.5.3/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
       14: from /home/himdel/.rbenv/versions/2.5.3/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/cli.rb:463:in `exec'
       13: from /home/himdel/.rbenv/versions/2.5.3/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/cli/exec.rb:28:in `run'
       12: from /home/himdel/.rbenv/versions/2.5.3/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/cli/exec.rb:74:in `kernel_load'
       11: from /home/himdel/.rbenv/versions/2.5.3/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/cli/exec.rb:74:in `load'
       10: from bin/rails:4:in `<top (required)>'
        9: from bin/rails:4:in `require'
        8: from /home/himdel/.rbenv/versions/2.5.3/lib/ruby/gems/2.5.0/gems/railties-5.0.7.2/lib/rails/commands.rb:18:in `<top (required)>'
        7: from /home/himdel/.rbenv/versions/2.5.3/lib/ruby/gems/2.5.0/gems/railties-5.0.7.2/lib/rails/commands/commands_tasks.rb:49:in `run_command!'
        6: from /home/himdel/.rbenv/versions/2.5.3/lib/ruby/gems/2.5.0/gems/railties-5.0.7.2/lib/rails/commands/commands_tasks.rb:78:in `console'
        5: from /home/himdel/.rbenv/versions/2.5.3/lib/ruby/gems/2.5.0/gems/railties-5.0.7.2/lib/rails/commands/console_helper.rb:9:in `start'
        4: from /home/himdel/.rbenv/versions/2.5.3/lib/ruby/gems/2.5.0/gems/railties-5.0.7.2/lib/rails/commands/console.rb:65:in `start'
        3: from (irb):4
        2: from /home/himdel/.rbenv/versions/2.5.3/lib/ruby/gems/2.5.0/gems/sys-filesystem-1.2.0/lib/sys/unix/sys/filesystem.rb:365:in `mount_point'
        1: from /home/himdel/.rbenv/versions/2.5.3/lib/ruby/gems/2.5.0/gems/sys-filesystem-1.2.0/lib/sys/unix/sys/filesystem.rb:365:in `stat'
Errno::EACCES (Permission denied @ rb_file_s_stat - /var/lib/postgresql/11/main
```

Reply:
(And apparently Fedora has `/var/lib/pgsql/` without x permission for regular users.)

Reply:
Internally it's doing a simple `File.stat` call. I guess the `df` command runs suid root.

Reply:
```
$df  '/var/lib/postgresql/11/main'
df: /var/lib/postgresql/11/main: Permission denied
```

I think the difference is just that you're not catching EACCESS , but the previous solution caught everything?

Reply:
TBH I don't think it's a good idea to change the access rights for the database service just to gather this information. Running as root is not a nice thing for developers and blocks us from the possibility to run MiQ as a non-root user in the future.

Reply:
Previous code was just using AwesomeSpawn.run!, but I don't see any explicit exception handling:

https://github.com/ManageIQ/manageiq-gems-pending/blob/master/lib/gems/pending/util/miq-system.rb#L112

Reply:
OK, I was thinking of the method removed here..

```
    rescue RuntimeError => err
      return nil if err.message.include?("does not exist")
```

Reply:
I can confirm that with the old implementation:

```
db_disk_size(data_directory) 
=> nil
```

Reply:
And the error being caught was:

```
=> #<RuntimeError: file /var/lib/postgresql/11/main does not exist>
```

Reply:
Errno::EACCESS is a subclass of SystemcallError, rescuing `RuntimeError` wouldn't affect this. I can update it to just return nil on  StandardError, though. That should work.

Reply:
Well, ok, guess it's time for a patch. Thanks @himdel